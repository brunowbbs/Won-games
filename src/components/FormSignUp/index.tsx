import Link from "next/link";
import {
  MdLockOutline,
  MdOutlineAccountCircle,
  MdOutlineEmail,
  MdOutlineMailOutline,
} from "react-icons/md";
import Button from "../Button";
import TextField from "../TextField";
import * as S from "./styles";

const FormSignUp = () => {
  return (
    <S.Wrapper>
      <form>
        <TextField
          name="name"
          placeholder="Name"
          type="text"
          icon={<MdOutlineAccountCircle />}
        />
        <TextField
          name="email"
          placeholder="Email"
          type="email"
          icon={<MdOutlineEmail />}
        />
        <TextField
          name="password"
          placeholder="Password"
          type="password"
          icon={<MdLockOutline />}
        />
        <TextField
          name="confirm-password"
          placeholder="Confirm password"
          type="confirm-password"
          icon={<MdLockOutline />}
        />

        <Button size="large" fullWidth>
          Sign up now
        </Button>
        <S.FormLink>
          Already have an account{" "}
          <Link href="/sign-in">
            <a>Sign in</a>
          </Link>
        </S.FormLink>
      </form>
    </S.Wrapper>
  );
};

export default FormSignUp;
