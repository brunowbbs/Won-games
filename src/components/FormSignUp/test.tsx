import { render, screen } from "@testing-library/react";
import { renderWithTheme } from "../../utils/tests/helpers";

import FormSignUp from ".";

describe("<FormSignIn/>", () => {
  it("Should render the form", () => {
    renderWithTheme(<FormSignUp />);

    expect(screen.getByPlaceholderText(/name/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Password")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Confirm password")).toBeInTheDocument();

    expect(
      screen.getByRole("button", { name: /sign up now/i })
    ).toBeInTheDocument();
  });

  it("Should render text and link to sign in", () => {
    renderWithTheme(<FormSignUp />);
    //Verificar o text
    expect(screen.getByText(/already have an account/i)).toBeInTheDocument();
    //Verificar o link
    expect(screen.getByRole("link", { name: /sign in/i })).toBeInTheDocument();
  });
});
