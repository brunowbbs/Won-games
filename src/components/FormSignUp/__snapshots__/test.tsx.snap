// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignUp /> should render text and link to sign in 1`] = `
<main
  class="sc-iBYQkv hzKDxw"
>
  <form>
    <div
      class="sc-gKPRtg jhOqha"
    >
      <div
        class="sc-gswNZR hXJmLO"
      >
        <div
          class="sc-eDvSVe eaWAWI"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33A7.95 7.95 0 014 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
            />
          </svg>
        </div>
        <input
          class="sc-dkrFOg hHzmDI"
          name="name"
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="sc-gKPRtg jhOqha"
    >
      <div
        class="sc-gswNZR hXJmLO"
      >
        <div
          class="sc-eDvSVe eaWAWI"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
            />
          </svg>
        </div>
        <input
          class="sc-dkrFOg hHzmDI"
          name="email"
          placeholder="Email"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="sc-gKPRtg jhOqha"
    >
      <div
        class="sc-gswNZR hXJmLO"
      >
        <div
          class="sc-eDvSVe eaWAWI"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 17c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm6-9h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM8.9 6c0-1.71 1.39-3.1 3.1-3.1s3.1 1.39 3.1 3.1v2H8.9V6zM18 20H6V10h12v10z"
            />
          </svg>
        </div>
        <input
          class="sc-dkrFOg hHzmDI"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="sc-gKPRtg jhOqha"
    >
      <div
        class="sc-gswNZR hXJmLO"
      >
        <div
          class="sc-eDvSVe eaWAWI"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 17c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm6-9h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM8.9 6c0-1.71 1.39-3.1 3.1-3.1s3.1 1.39 3.1 3.1v2H8.9V6zM18 20H6V10h12v10z"
            />
          </svg>
        </div>
        <input
          class="sc-dkrFOg hHzmDI"
          name="confirm-password"
          placeholder="Confirm password"
          type="confirm-password"
          value=""
        />
      </div>
    </div>
    <button
      class="sc-bcXHqe PsMiZ"
    >
      <span>
        Sign up now
      </span>
    </button>
    <div
      class="sc-ftTHYK kIgqIw"
    >
      Already have an account
       
      <a
        href="/sign-in"
      >
        Sign in
      </a>
    </div>
  </form>
</main>
`;
